File: backend/src/main/java/com/java/NBE4_5_1_7/domain/chat/repository/ChatMessageRepository.java
package com.java.NBE4_5_1_7.domain.chat.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.java.NBE4_5_1_7.domain.chat.entity.ChatMessage;

public interface ChatMessageRepository extends JpaRepository<ChatMessage, Long> {
	List<ChatMessage> findByChatRoomId(Long chatRoomId);
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/chat/repository/ChatMessageRepository.kt
package com.java.NBE4_5_3_7.domain.chat.repository

import com.java.NBE4_5_3_7.domain.chat.entity.ChatMessageEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime

interface ChatMessageRepository : JpaRepository<ChatMessageEntity, Long> {
    fun existsByRoomIdAndSenderAndContentAndTimestamp(
        roomId: Long,
        sender: String,
        content: String,
        timestamp: LocalDateTime
    ): Boolean

}

